app/invitations/api.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+from fastapi import APIRouter, BackgroundTasks, Depends
+from fastapi import status as http_status
+from fastapi.security import HTTPBearer
+from starlette.responses import JSONResponse
+
+from app import settings
+from app.core.auth_jwt_bearer import JWTBearer
+from app.core.exceptions import OptScaleAPIResponseError, handle_exception
+from app.invitations.model import (
+    DeclineInvitation,
+    RegisteredInvitedUserResponse,
+    RegisterInvitedUser,
+)
+from app.invitations.services.external_services import (
+    register_invited_user_on_optscale,
+    remove_user,
+)
+from app.optscale_api.auth_api import OptScaleAuth
+from app.optscale_api.helpers.auth_tokens_dependency import get_auth_client
+from app.optscale_api.invitation_api import OptScaleInvitationAPI
+from app.optscale_api.orgs_api import OptScaleOrgAPI
+from app.optscale_api.users_api import OptScaleUserAPI
+
+# HTTPBearer scheme to parse Authorization header
+bearer_scheme = HTTPBearer()
+logger = logging.getLogger(__name__)
+router = APIRouter()
+
+
+@router.post(
+    path="/users",
+    status_code=http_status.HTTP_201_CREATED,
+    response_model=RegisteredInvitedUserResponse,
+)
+async def register_invited(data: RegisterInvitedUser):
+    try:
+        response = await register_invited_user_on_optscale(
+            email=str(data.email),
+            display_name=data.display_name,
+            password=data.password,
+        )
+        return JSONResponse(
+            status_code=response.get("status_code", http_status.HTTP_201_CREATED),
+            content=response.get("data", {}),
+        )
+
+    except OptScaleAPIResponseError as error:
+        handle_exception(error=error)
+
+
+@router.post(
+    path="/users/invites/{invite_id}/decline",
+    status_code=http_status.HTTP_200_OK,
+    dependencies=[Depends(JWTBearer())],
```

Comment:
The `decline` is consumed by the UI so you will get a user token here not a JWT token.

Reply:
True.